models/product.go

```diff
@@ -4,6 +4,14 @@ import (
 	"github.com/google/uuid"
 )
 
+type ProductData struct {
+	ID      uuid.UUID
+	Name    string
+	Public  bool
+	Assets  Asset
+	Details Asset
```

Comment:
Are details assets?

Reply:
Yeah. So settings, details, assets are all stored with asset types, since all are a UUID + a json field from DB perspective. The only reason I actually differentiate is because I have separate tables for each, to avoid having too much data in a single table.
So there are four tables now: product_details, product_assets, user_assets, user_settings. And all are represented with an asset struct in the backend code.

If you think there is a better name for the struct based on this and you have a suggestion. I can change.